universal-login-sdk/lib/core/services/GasModeService.ts

```diff
@@ -1,21 +1,33 @@
 import {utils} from 'ethers';
-import {GasMode, TokensPrices} from '@universal-login/commons';
-import {GasPriceOracle} from '../../integration/ethereum/gasPriceOracle';
+import {GasMode, TokensPrices, TokensValueConverter, ObservedCurrency, ensureNotNull} from '@universal-login/commons';
 import {TokensDetailsStore} from './TokensDetailsStore';
+import {GasPriceOracle} from '../../integration/ethereum/gasPriceOracle';
 import {PriceObserver} from '../observers/PriceObserver';
 
 export class GasModeService {
   constructor(
     private tokensStore: TokensDetailsStore,
     private gasPriceOracle: GasPriceOracle,
-    private priceObserver: PriceObserver
-    ) {}
+    private priceObserver: PriceObserver,
+    private tokensValueConverter: TokensValueConverter
+  ) {}
+
+  private getTokenPriceInversed(tokenPrices: TokensPrices, symbol: string) {
+    const multiplier = tokenPrices.ETH[symbol as any as ObservedCurrency];
+    ensureNotNull(multiplier, Error, 'Invalid fee token');
+    return multiplier;
+  }
 
   private createMode(name: string, gasPrice: utils.BigNumber, tokensPrices: TokensPrices): GasMode {
-    return {name, gasOptions: this.tokensStore.tokensDetails.map((tokenDetails) => ({
-      token: tokenDetails,
-      gasPrice,
-    }))};
+    return {
+      name, gasOptions: this.tokensStore.tokensDetails.map((tokenDetails) => {
```

Comment:
It seems to me that:
TokenDetails.symbol should be of type ObservedCurrency

Reply:
Another considerations:
- Why `ObserveredCurrency`, let's just to `Currency`. One global type in `commons`
- we could do subtype, e.g. `type Currency = FiatCurrency | CryptoCurrency;` (if needed)